darwin/Platform.c

```diff
@@ -464,10 +469,65 @@ bool Platform_getDiskIO(DiskIOData* data) {
    return true;
 }
 
+/* Caution: Given that interfaces are dynamic, and it is not possible to get statistics on interfaces that no longer exist,
+   if some interface disappears between the time of two samples, the values of the second sample may be lower than those of
+   the first one. */
 bool Platform_getNetworkIO(NetworkIOData* data) {
-   // TODO
-   (void)data;
-   return false;
+   int mib[6] = {CTL_NET,
+      PF_ROUTE, /* routing messages */
+      0, /* protocal number, currently always 0 */
+      0, /* select all address families */
+      NET_RT_IFLIST2, /* interface list with addresses */
+      0};
+
+   for (unsigned int retry = 0; retry < 4; retry++) {
+      size_t len;
+
+      /* Determine len */
+      if (sysctl(mib, ARRAYSIZE(mib), NULL, &len, NULL, 0) < 0 || len == 0)
+         return false;
+
+      len += 16 * retry * retry * sizeof(struct if_msghdr2);
```

Comment:
What does this `16 * retry * retry` come from?

Reply:
That's for exponential back-off to over-allocate in case of rapid fluctuations. Not strictly necessary for network stuff, but doesn't hurt either.